src/include/rtt_if.h

```diff
@@ -29,7 +29,9 @@
 
 /* rtt i/o to terminal */
 
-#if PC_HOSTED == 0
+#include "platform.h"
+
+#if PC_HOSTED == 0 && NO_LIBOPENCM3 != 1
```

Comment:
If memory serves, NO_LIBOPENCM3 is not defined if we're using locm3, so could this be done as `#if PC_HOSTED == 0 && !defined(NO_LIBOPENCM3)`?

Reply:
Perhaps this is not a good approach, because it's not actually defined anywhere. It only exists in the `Makefile`, and is never passed as a build flag.

I do define a flag in Farpatch, but perhaps that approach is too platform-specific.

I'm open to suggestions for other approaches. For example, defining a dummy `struct _usbd_device;` and then having the function accept a `struct _usbd_device *` (which never actually gets defined).

Reply:
A comment in the header would be perfectly acceptable. BMP itself does not `-DNO_LIBOPENCM3` so hence asking to use the `defined()` check rather than an integral value check.

Good question re ccache, unfortunately we don't know enough about the tool to make any suggestions on that score

Reply:
Alright, I've modified it and verified that it still builds for the three targets in question.